Kuroba/app/src/main/java/com/github/adamantcheese/chan/report/ReportProblemController.kt

```diff
@@ -0,0 +1,154 @@
+            reportActivityAttachLogsButton = findViewById(R.id.report_activity_attach_logs_button)
+            reportActivityLogsText = findViewById(R.id.report_activity_logs_text)
+            reportActivityCancel = findViewById(R.id.report_activity_cancel)
+            reportActivitySendReport = findViewById(R.id.report_activity_send_report)
+        }
+
+        presenter.onCreate(this)
+        compositeDisposable = CompositeDisposable()
+
+        val logs = LogsController.loadLogs()
+        if (logs != null) {
+            reportActivityLogsText.setText(logs)
+        }
+
+        reportActivityAttachLogsButton.setOnCheckedChangeListener { _, isChecked ->
+            reportActivityLogsText.isEnabled = isChecked
+        }
+        reportActivityCancel.setOnClickListener { onFinishedListener?.invoke() }
+        reportActivitySendReport.setOnClickListener { onSendReportClick() }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+
+        onFinishedListener = null
+        presenter.onDestroy()
+        compositeDisposable.dispose()
+    }
+
+    fun setOnFinishedCallback(callback: () -> Unit) {
+        this.onFinishedListener = callback
+    }
+
+    private fun onSendReportClick() {
+        val title = reportActivityProblemTitle.text?.toString() ?: ""
+        val description = reportActivityProblemDescription.text?.toString() ?: ""
+        val logs = reportActivityLogsText.text?.toString() ?: ""
+
+        if (title.isEmpty()) {
+            reportActivityProblemTitle.error =
+                    context.getString(R.string.report_activity_title_cannot_be_empty_error)
+            return
+        }
+
+        if (description.isEmpty()) {
+            reportActivityProblemDescription.error =
+                    context.getString(R.string.report_activity_description_cannot_be_empty_error)
+            return
+        }
+
+        if (reportActivityAttachLogsButton.isChecked && logs.isEmpty()) {
+            reportActivityLogsText.error =
+                    context.getString(R.string.report_activity_logs_are_empty_error)
+            return
+        }
+
+        val logsParam = if (!reportActivityAttachLogsButton.isChecked) {
+            null
+        } else {
+            logs
+        }
+
+        showProgressDialog()
+
+        presenter.sendReport(title, description, logsParam)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnTerminate { hideProgressDialog() }
+                .subscribe({ result ->
+                    handleResult(result)
+                }, { error ->
+                    Logger.e(TAG, "Error", error)
+
+                    val errorMessage = error.message ?: "No error message"
+                    AndroidUtils.showToast(
+                            context.getString(R.string.report_activity_error_while_trying_to_send_report,
```

Comment:
Ya did it again. 
Stop that. 🤕🗞️

Reply:
This was written before you pointed it out the first time. I'll fix it.

Reply:
Done.